tensorboard/webapp/metrics/views/card_renderer/scalar_card_container.ts

```diff
@@ -603,25 +604,41 @@ export class ScalarCardContainer implements CardRenderer, OnInit, OnDestroy {
       this.minMaxSteps$,
       this.store.select(getMetricsStepSelectorEnabled),
       this.store.select(getMetricsRangeSelectionEnabled),
-    ]).subscribe(
-      ([{minStep, maxStep}, enableStepSelector, stepSelectorRangeEnabled]) => {
-        this.stepSelectorTimeSelection$.next(
-          enableStepSelector
-            ? {
-                start: {step: minStep},
-                end: stepSelectorRangeEnabled ? {step: maxStep} : null,
-              }
-            : null
-        );
-      }
-    );
+    ]).subscribe(([minMaxStep, enableStepSelector, rangeSelectionEnabled]) => {
+      this.stepSelectorTimeSelection$.next(
+        this.computeStepSelectorTimeSelection(
```

Comment:
I do not think a utility function is needed here.

Reply:
This makes it a lot easier to test

Reply:
Not my favorite reason to add an abstraction. We are also now missing the logic that this is called when minMaxStep, enableStepSelector, or rangeSelectionEnabled are changed. And adding those tests are just as difficult as testing this without the function. I will accept it because I want this fix in and it is not that bad.